src/components/Onboarding/steps.ts

```diff
@@ -0,0 +1,261 @@
+/* eslint-disable max-lines */
+import { Translate } from 'next-translate';
+import { Step as BaseStep } from 'react-joyride';
+
+// import OnboardingStep from './OnboardingStep';
+
+import OnboardingGroup from '@/types/OnboardingGroup';
+import { getSurahNavigationUrl } from '@/utils/navigation';
+
+type Step = BaseStep & {
+  showNextButton?: boolean;
+  showPrevButton?: boolean;
+};
+
+// eslint-disable-next-line react-func/max-lines-per-function
+export const checklistIndexToOnboardingSteps = (
+  t: Translate,
+  component: React.ElementType,
+): Record<
+  OnboardingGroup,
+  {
+    title: string;
+    description: string;
+    step: Partial<Step>;
+  }[]
+> => {
+  const commonStepOptions: Partial<Step> = {
+    content: '',
+    placement: 'auto',
+    disableBeacon: true,
+    hideBackButton: true,
+    disableOverlay: false,
+    disableOverlayClose: true,
+    tooltipComponent: component as any,
+    floaterProps: {
+      hideArrow: true,
+    },
+    spotlightPadding: 10,
+  };
+
+  const getDetails = (key: string) => ({
+    title: t(`onboarding:steps.${key}.title`),
+    description: t(`onboarding:steps.${key}.description`),
+  });
+
+  return {
+    [OnboardingGroup.HOMEPAGE]: [
```

Comment:
not a very strong opinion but I think we should break this object into either multiple ones OR multiple files for easier readability and future refactoring.